packages/arch3-core/src/modules/authz/tx.ts

```diff
@@ -0,0 +1,108 @@
+import { AminoConverters } from "@cosmjs/stargate";
+import { GenericAuthorization } from "cosmjs-types/cosmos/authz/v1beta1/authz";
+import { MsgGrant, MsgRevoke } from "cosmjs-types/cosmos/authz/v1beta1/tx";
+import { SendAuthorization } from "cosmjs-types/cosmos/bank/v1beta1/authz";
+import { Timestamp } from "cosmjs-types/google/protobuf/timestamp";
+import { fromJsonTimestamp, fromTimestamp } from "cosmjs-types/helpers";
+
+export function createAuthzAminoConverters(): AminoConverters {
+  return {
+    [MsgGrant.typeUrl]: {
+      aminoType: "cosmos-sdk/MsgGrant",
+      toAmino: ({ granter, grantee, grant }) => {
+        if (!grant || !grant.authorization) {
+          throw new Error(`Unsupported grant type: '${grant?.authorization?.typeUrl}'`);
+        }
+        let authorizationValue;
+        switch (grant?.authorization?.typeUrl) {
+          case GenericAuthorization.typeUrl: {
+            const generic = GenericAuthorization.decode(grant.authorization.value);
+            authorizationValue = {
+              type: "cosmos-sdk/GenericAuthorization",
+              value: {
+                msg: generic.msg,
+              },
+            };
+            break;
+          }
+          case SendAuthorization.typeUrl: {
+            const spend = SendAuthorization.decode(grant.authorization.value);
+            authorizationValue = {
+              type: "cosmos-sdk/SendAuthorization",
+              value: {
+                spend_limit: spend.spendLimit,
```

Comment:
Since it's in `arch3.js` now, we need to support all the possible fields in the converters, e.g. `allow_list` in this case